apps/dc_tools/odc/apps/dc_tools/index_from_tar.py

```diff
@@ -1,13 +1,143 @@
 from odc.io.tar import tar_doc_stream, tar_mode
 from odc.io.timer import RateEstimator
-from odc.index import from_yaml_doc_stream
+from pyproj import CRS, Transformer
+
+
+KNOWN_CONSTELLATIONS = [
+    'sentinel-2'
+]
+
+
+def _stac_product_lookup(item):
+    product_label = item['id']
+    product_name = item['properties']['platform']
+    region_code = None
+
+    constellation = item['properties']['constellation']
+
+    if constellation in KNOWN_CONSTELLATIONS:
+        if constellation == 'sentinel-2':
+            product_label = item['properties']["sentinel:product_id"]
+            product_split = product_label.split("_")
+            product_name = f"{product_split[0]}_{product_split[1]}"
+            region_code = "{}{}{}".format(
+                str(item['properties']["proj:epsg"])[-2:],
+                item['properties']["sentinel:latitude_band"],
+                item['properties']["sentinel:grid_square"]
+            )
+
+    return product_label, product_name, region_code
+
+
+def _get_stac_bands(item, default_grid='g10.0m'):
+    bands = {}
+
+    grids = {}
+
+    assets = item['assets']
+
+    for asset_name, asset in assets.items():
+        # Ignore items that are not actual COGs
+        if asset['type'] not in ["image/tiff; application=geotiff; profile=cloud-optimized"]:
+            continue
+
+        transform = asset['proj:transform']
+        grid = "g{}m".format(transform[0])
+
+        if grid not in grids:
+            grids[grid] = {
+                'shape': asset['proj:shape'],
+                'transform': asset['proj:transform']
+            }
+
+        band_info = {
+            'path': os.path.split((asset['href']))[1]
+        }
+
+        if grid != default_grid:
+            band_info['grid'] = grid
+
+        bands[asset_name] = band_info
+
+    grids['default'] = grids[default_grid]
+    del grids[default_grid]
+
+    return bands, grids
+
+
+def geographic_to_projected(geometry, target_srs):
+    wgs = CRS.from_epsg(4326)
+    projection = CRS.from_epsg(target_srs)
+
+    transformer = Transformer.from_crs(wgs, projection)
+
+    new_geometry = deepcopy(geometry)
+    new_geometry['coordinates'][0] = [
+        transformer.transform(p[0], p[1]) for p in new_geometry['coordinates'][0]]
+
+    return new_geometry
+
+
+def stac_transform(input_stac):
+    """takes in a raw STAC 1.0 dictionary and returns an ODC dictionary
+    """
+    print("Starting STAC transform")
+
+    # Dodgy lookup
+    product_label, product_name, region_code = _stac_product_lookup(input_stac)
+
+    # Make a proper deterministic UUID
+    deterministic_uuid = str(odc_uuid("sentinel2_stac_process", "1.0.0", [product_label]))
+
+    # Get grids and bands
+    bands, grids = _get_stac_bands(input_stac)
+
+    stac_odc = {
+        '$schema': 'https://schemas.opendatacube.org/dataset',
+        'id': deterministic_uuid,
+        'crs': "epsg:{}".format(input_stac['properties']['proj:epsg']),
```

Comment:
there are so many `input_stac['properties']` in here, I would

```python
props = input_stac['properties']
...
use(props['proj:epsg'])
```

This will read better, be safer and more performant too.